src/OSPSuite.FuncParserNative/include/FuncParser/FuncParser.h

```diff
@@ -61,18 +61,18 @@ class FuncParser
 		const Constants & GetConstants () const;
 	
 	private:
-		bool IsBracketed (const std::string & SubExpr);
-		void RemoveBrackets (std::string & SubExpr);
+		bool IsBracketed (const std::string & SubExpr) const;
+      std::string RemoveBrackets (const std::string & SubExpr) const;
 	
 	public:
 		FuncNode * Parse (const std::string & ParsedString, const std::vector < std::string > & VariableNames, const std::vector < std::string > & ParameterNames, bool CaseSensitive, bool LogicOperatorsAllowed, double ComparisonTolerance, bool LogicalNumericMixAllowed);
 	
 	private:
-		void EvalExpression (FuncNode * SubNode, std::string SubExpr, enmLevelOfAbstraction LevelOfAbstraction);
-		void EvalNOTOperand (FuncNode * SubNode, std::string SubExpr);
-		void EvalComparison (FuncNode * SubNode, std::string SubExpr);
-		void EvalFactor (FuncNode * SubNode, std::string SubExpr);
-		void EvalIF (FuncNode * SubNode, std::string SubExpr);
+		void EvalExpression (FuncNode * SubNode, const std::string & SubExpr, enmLevelOfAbstraction LevelOfAbstraction);
+		void EvalNOTOperand (FuncNode * SubNode, const std::string & SubExpr);
+		void EvalComparison (FuncNode * SubNode, const std::string & SubExpr);
+		void EvalFactor (FuncNode * SubNode, const std::string & SubExpr);
+		void EvalIF (FuncNode * SubNode, const std::string & SubExpr);
```

Comment:
This is in fact the only relevant change. All those functions call each other in a circle (EvalA=>EvalB=>...=>EvalX=>EvalA), and the substring to analyze was passed as a copy - just because in one of hte functions it was modified by the `RemoveBrackets` function.

I changed this, so that `RemoveBrackets` returns a modified copy of the string which is used from there on.
This allowed to pass all the substrings as a const reference now, so no string copying.

However no speedup with this in the multithreaded runs :(

Reply:
hum...maybe memory allocation will be better...